src/gui_common/charts/line/LineChart.cs

```diff
@@ -0,0 +1,591 @@
+                };
+
+                var itemId = 0;
+
+                dropDown.Popup.HideOnCheckableItemSelection = false;
+
+                foreach (var data in DataSets)
+                {
+                    // Use the default icon as a fallback if the data icon texture hasn't been set already
+                    data.Value.IconTexture = data.Value.IconTexture == null ?
+                        defaultIconLegendTexture :
+                        data.Value.IconTexture;
+
+                    // Use the DataColor as the icon's color if using the default icon
+                    var colorToUse = data.Value.IconTexture == defaultIconLegendTexture ?
+                        data.Value.DataColor :
+                        new Color(1, 1, 1);
+
+                    dropDown.AddItem(data.Key, itemId, true, colorToUse, data.Value.IconTexture);
+
+                    // Set initial item check state
+                    if (data.Value.Draw)
+                        dropDown.Popup.SetItemChecked(dropDown.Popup.GetItemIndex(itemId), true);
+
+                    itemId++;
+                }
+
+                legendContainer.AddChild(dropDown);
+
+                dropDown.Popup.Connect("index_pressed", this, nameof(DropDownLegendItemSelected),
+                    new Array { dropDown });
+
+                break;
+            }
+
+            default:
+                throw new Exception("Invalid legend display mode");
+        }
+    }
+
+    /// <summary>
+    ///   Draw the chart visuals. Mainly used by the Drawer node to connect its 'draw()' signal here.
+    /// </summary>
+    private void RenderChart()
+    {
+        if (DataSets.Count <= 0)
+            return;
+
+        DrawOrdinateLines();
+        DrawLineSegments();
+    }
+
+    /// <summary>
+    ///   Draw columns of lines going horizontal
+    /// </summary>
+    private void DrawOrdinateLines()
+    {
+        for (int i = 0; i < YAxisTicks; i++)
+        {
+            var value = Mathf.Round(i * (MaxValues.y - MinValues.y) /
+                (YAxisTicks - 1) + MinValues.y);
+
+            drawArea.DrawTextureRect(hLineTexture, new Rect2(new Vector2(
+                0, ConvertToYCoordinate(value)), RectSize.x, 1), false, new Color(1, 1, 1, 0.3f));
+        }
+    }
+
+    /// <summary>
+    ///   Connect the dataset points with line segments
+    /// </summary>
+    private void DrawLineSegments()
+    {
+        foreach (var data in DataSets)
+        {
+            // Setup the points
+            data.Value.DataPoints.ForEach(point =>
+            {
+                point.Coordinate = ConvertToCoordinate(point.Value);
+
+                if (!point.IsInsideTree())
+                    drawArea.AddChild(point);
+
+                var pointHasZeroValue = point.Value.x <= 0 || point.Value.y <= 0;
+
+                point.Draw = pointHasZeroValue ? false : true;
+
+                // Keep draw if this flag is set to true
```

Comment:
Grammatical error in this sentence?

Reply:
Probably, I must've dozed off writing this. err maybe this can be:
```suggestion
                // Let the markers at zero coordinates be drawn if this flag is set to true
```